src/Shouldly.Tests/ShouldNotThrowAsync/FuncOfTaskAsyncScenario.cs

```diff
@@ -0,0 +1,47 @@
+using NUnit.Framework;
+#if net40
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Shouldly.Tests.ShouldNotThrowAsync
+{
+    [TestFixture]
+    public class FuncOfTaskAsyncScenario
+    {
+        [Test]
+        public void ShouldThrowAWobbly()
+        {
+            try
+            {
+                var task = Task.Factory.StartNew(() => {
+                                                           throw new RankException();
+                },
+                    CancellationToken.None, TaskCreationOptions.None,
+                    TaskScheduler.Default);
+                Should.NotThrowAsync(task, TimeSpan.FromMilliseconds(2000), () => "Some additional context").Wait();
+                
+            }
+            catch (AggregateException e)
+            {
+                var inner = e.Flatten().InnerException;
+                var ex = inner.ShouldBeOfType<RankException>();
```

Comment:
Should be of type `ShouldlyAssertException` and the inner exception of that exception should be the RankException. I think that is what the other methods do